docs/labels.md

```diff
@@ -11,50 +10,93 @@ key/value labels set on it, with at most one label with a particular key.
 ```
-key1
-key1 = value11
-key1 != value11
-key1 in (value11, value12, ...)
-key1 not in (value11, value12, ...)
+environment in (production, qa)
+tier notin (frontend, backend)
+partition
 ```
+The first example selects all resources with key equal to `environment` and value equal to `production` or `qa`.
+The second example selects all resources with key equal to `tier` and value other than `frontend` and `backend`.
+The third example selects all resources including a label with key `partition`; no values are checked.
+Similary the comma separator acts as an _AND_ operator for example filtering resource with a `partition` key (not matter the value) and with `environment` different than  `qa`. For example: `partition,environment notin (qa)`.
+The _set-based_ label selector is a general form of equality since `environment=production` is equivalent to `environment in (production)`; similarly for `!=` and `notin`.
+
+_Set-based_ requirements can be mixed with _equality-based_ requirements. For example: `partition in (customerA, customerB),environment!=qa`.
+
+
+## API
+
+LIST and WATCH operations may specify label selectors to filter the sets of objects returned using a query parameter. Both requirements are permitted:
+   - _equality-based_ requirements: `?labels=key1%3Dvalue1,key2%3Dvalue2`
+   - _set-based_ requirements: `?labels=key+in+%28value1%2Cvalue2%29%2Ckey2+notin+%28value3`
+
+Kubernetes also currently supports two objects that use label selectors to keep track of their members, `service`s and `replicationController`s:
+- `service`: A [service](/docs/services.md) is a configuration unit for the proxies that run on every worker node.  It is named and points to one or more pods.
+- `replicationController`: A [replication controller](/docs/replication-controller.md) ensures that a specified number of pod "replicas" are running at any one time.
+
+The set of pods that a `service` targets is defined with a label selector. Similarly, the population of pods that a `replicationController` is monitoring is also defined with a label selector. For management convenience and consistency, `services` and `replicationControllers` may themselves have labels and would generally carry the labels their corresponding pods have in common.
+
+
+## Future developments
 
-LIST and WATCH operations may specify label selectors to filter the sets of objects returned using a query parameter: `?labels=key1%3Dvalue1,key2%3Dvalue2,...`.
+Concerning API: we may extend such filtering to DELETE operations in the future.
 
-The `service` and `replicationController` kinds of objects use selectors to match sets of pods that they operate on.
+We'll eventually index and reverse-index labels for efficient queries and watches, use them to sort and group in UIs and CLIs, etc. We don't want to pollute labels with non-identifying, especially large and/or structured, data. Non-identifying information should be recorded using [annotations](/docs/annotations.md).
```

Comment:
Please put the latter 2 sentences about annotations to a paragraph on their own, just above the Motivation section.